src/util/RtreeBuild.cpp

```diff
@@ -0,0 +1,1133 @@
+  archive << node;
+  nodesOfs.write(" ", 1);
+
+  return pos;
+}
+
+std::optional<Rtree::BoundingBox> GetBoundingBoxFromWKT(const std::string& wkt) {
+  /**
+     * Parse the wkt literal in a way, that only the relevant data for the rtree gets read in.
+   */
+  bool lookingForX = true;
+  bool readingDouble = false;
+  std::string currentDouble;
+
+  double minX = -1;
+  double maxX = -1;
+  double minY = -1;
+  double maxY = -1;
+
+  for (char c : wkt) {
+    if (isdigit(c)) {
+      readingDouble = true;
+      currentDouble += c;
+    } else if (c == '.') {
+      readingDouble = true;
+      currentDouble += '.';
+    } else if (c == ' ') {
+      if (readingDouble && lookingForX) {
+        // x is completely read in
+        readingDouble = false;
+        lookingForX = false;
+        double x;
+        try {
+          x = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (x < minX || minX == -1) {
+          minX = x;
+        }
+
+        if (x > maxX) {
+          maxX = x;
+        }
+      }
+    } else {
+      if (readingDouble && !lookingForX) {
+        // y is completely read in
+        readingDouble = false;
+        lookingForX = true;
+        double y;
+        try {
+          y = std::stod(currentDouble);
+        } catch(...) {
+          return { };
+        }
+        currentDouble = "";
+        if (y < minY || minY == -1) {
+          minY = y;
+        }
+
+        if (y > maxY) {
+          maxY = y;
+        }
+      }
+    }
+  }
+
+  return { Rtree::createBoundingBox(minX, minY, maxX, maxY) };
+}
+
+std::optional<Rtree::BoundingBox> Rtree::ConvertWordToRtreeEntry(const std::string& wkt) {
+  /**
+     * Convert a single wkt literal to a boundingbox.
+   */
+  std::optional<Rtree::BoundingBox> boundingBox;
+
+  /* Get the bounding box(es) of either a multipolygon, polygon or a linestring */
+  std::size_t posWKTStart = wkt.find("MULTIPOLYGON(((") + 14;
+  std::size_t posWKTEnd = wkt.find(")))", posWKTStart);
+  if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+    std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+    boundingBox = GetBoundingBoxFromWKT(newWkt);
+  } else {
+    posWKTStart = wkt.find("POLYGON((") + 8;
+    posWKTEnd = wkt.find("))", posWKTStart);
+    if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+      std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+      boundingBox = GetBoundingBoxFromWKT(newWkt);
+    } else {
+      posWKTStart = wkt.find("LINESTRING(") + 10;
+      posWKTEnd = wkt.find(')', posWKTStart);
+      if (posWKTStart != std::string::npos && posWKTEnd != std::string::npos) {
+        std::string newWkt = wkt.substr(posWKTStart, posWKTEnd - posWKTStart + 1);
+        boundingBox = GetBoundingBoxFromWKT(newWkt);
+      } else {
+        return { };
+      }
+    }
+  }
```

Comment:
These functions should be in a separate file. And if your `getBoundingBox` function can handle the full strings, then this function is basically only
`starts_with("POLYGON") || starts_with(LINESTRING) || starts_with("MULTIPOLYGON").